backend/pkg/api/connect/integration/transform_test.go

```diff
@@ -0,0 +1,95 @@
+package integration
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"connectrpc.com/connect"
+	"github.com/stretchr/testify/assert"
+	rqr "github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kadm"
+
+	v1alpha1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1"
+	v1alpha1connect "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1/dataplanev1alpha1connect"
+)
+
+func spawnTopic(ctx context.Context, partitionCount int32, client v1alpha1connect.TopicServiceClient, topic string) error {
+	_, err := client.CreateTopic(ctx, connect.NewRequest(&v1alpha1.CreateTopicRequest{
+		Topic: &v1alpha1.CreateTopicRequest_Topic{
+			Name:           topic,
+			PartitionCount: &partitionCount,
+			Configs: []*v1alpha1.CreateTopicRequest_Topic_Config{
+				{
+					Name:  "cleanup.policy",
+					Value: kadm.StringPtr("compact"),
+				},
+			},
+		},
+	}))
+	return err
+}
+
+func despawnTopic(ctx context.Context, client v1alpha1connect.TopicServiceClient, topic string) error {
+	_, err := client.DeleteTopic(ctx, connect.NewRequest(&v1alpha1.DeleteTopicRequest{
+		Name: topic,
+	}))
+	return err
+}
+
+func (s *APISuite) TestCreateTransform() {
+	t := s.T()
+
+	require := rqr.New(t)
+	// assert := assert.New(t)
+
+	t.Run("create transform with valid request (connect-go)", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
+		defer cancel()
+
+		topicClient := v1alpha1connect.NewTopicServiceClient(http.DefaultClient, s.httpAddress())
+		require.NoError(spawnTopic(ctx, 2, topicClient, "wasm-tfm-create-test-i"))
+		require.NoError(spawnTopic(ctx, 2, topicClient, "wasm-tfm-create-test-o"))
+
+		buf, err := os.ReadFile(os.Getenv("WASM_BINARY_PATH"))
+		require.NoError(err)
+
+		transformClient := v1alpha1connect.NewTransformServiceClient(http.DefaultClient, s.httpAddress())
+		_, err = transformClient.DeployTransform(ctx, connect.NewRequest(&v1alpha1.DeployTransformRequest{
+			Name:             "foo",
+			WasmBinary:       buf,
+			InputTopicName:   "wasm-tfm-create-test-i",
+			OutputTopicNames: []string{"wasm-tfm-create-test-o"},
+			Environment:      nil,
+		}))
+		require.NoError(err)
```

Comment:
minor nit, but maybe this should be an assert, since _this is the test_?

Reply:
I think some wires got crossed as that test file was very much not ready for review yet (as you noticed)! I had pushed it up purely to ask some questions about handling the binary from a couple people. 

The complete testfile is now up and available for review.